From 691f597fbe6f373366638209b14449162ed95496 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin2.jansa@lgepartner.com>
Date: Fri, 24 Jan 2025 15:08:06 +0000
Subject: [PATCH] CMakeLists.txt: allow to disable building examples to fix
 build with boost-1.87

:Release Notes:
WITH_EXAMPLES option was declared but wasn't uses anywhere.
Examples fail to build with boost-1.87.0.

:Detailed Notes;
Fails with:
http://gecko.lge.com:8000/Errors/Details/1086933

FAILED: example/tools/CMakeFiles/edgeai-example-tools.dir/WSServerSession.cpp.o
TOPDIR/BUILD/work/o22-webos-linux/edgeai-vision/1.1.0-85/recipe-sysroot-native/usr/bin/aarch64-webos-linux/aarch64-webos-linux-g++ --sysroot=TOPDIR/BUILD/work/o22-webos-linux/edgeai-vision/1.1.0-85/recipe-sysroot -DBOOST_ATOMIC_DYN_LINK -
DBOOST_ATOMIC_NO_LIB -DBOOST_DATE_TIME_DYN_LINK -DBOOST_DATE_TIME_NO_LIB -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_FILESYSTEM_NO_LIB -DBOOST_IOSTREAMS_DYN_LINK -DBOOST_IOSTREAMS_NO_LIB -DBOOST_RANDOM_DYN_LINK -DBOOST_RANDOM_NO_LIB -DBOOST_REGEX
_DYN_LINK -DBOOST_REGEX_NO_LIB -DBOOST_SYSTEM_DYN_LINK -DBOOST_SYSTEM_NO_LIB -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_NO_LIB -DEDGEAI_VISION_EXTENSION_REGISTRY_PATH=\"/tmp/edgeai_extension_registry.json\" -DENABLE_DEBUG -DLOG_APPLICATION=\"
AIF\" -Dedgeai_example_tools_EXPORTS -ITOPDIR/BUILD/work/o22-webos-linux/edgeai-vision/1.1.0-85/git/include -ITOPDIR/BUILD/work/o22-webos-linux/edgeai-vision/1.1.0-85/git/third_party -ITOPDIR/BUILD/work/o22-webos-linux/edgeai-vision/1.1.0
-85/git/example/tools -ITOPDIR/BUILD/work/o22-webos-linux/edgeai-vision/1.1.0-85/recipe-sysroot/usr/include/opencv4 -fstack-protector-all  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security -Werror=return-type -fun
wind-tables  --sysroot=TOPDIR/BUILD/work/o22-webos-linux/edgeai-vision/1.1.0-85/recipe-sysroot  -O2 -g -fcanon-prefix-map  -fmacro-prefix-map=TOPDIR/BUILD/work/o22-webos-linux/edgeai-vision/1.1.0-85/git=/usr/src/debug/edgeai-vision/1.1.0-
85  -fdebug-prefix-map=TOPDIR/BUILD/work/o22-webos-linux/edgeai-vision/1.1.0-85/git=/usr/src/debug/edgeai-vision/1.1.0-85  -fmacro-prefix-map=TOPDIR/BUILD/work/o22-webos-linux/edgeai-vision/1.1.0-85/build=/usr/src/debug/edgeai-vision/1.1.
0-85  -fdebug-prefix-map=TOPDIR/BUILD/work/o22-webos-linux/edgeai-vision/1.1.0-85/build=/usr/src/debug/edgeai-vision/1.1.0-85  -fdebug-prefix-map=TOPDIR/BUILD/work/o22-webos-linux/edgeai-vision/1.1.0-85/recipe-sysroot=  -fmacro-prefix-map
=TOPDIR/BUILD/work/o22-webos-linux/edgeai-vision/1.1.0-85/recipe-sysroot=  -fdebug-prefix-map=TOPDIR/BUILD/work/o22-webos-linux/edgeai-vision/1.1.0-85/recipe-sysroot-native=  -fmacro-prefix-map=TOPDIR/BUILD/work/o22-webos-linux/edgeai-vis
ion/1.1.0-85/recipe-sysroot-native=  -pipe -fvisibility-inlines-hidden  -std=gnu++17 -fPIC -MD -MT example/tools/CMakeFiles/edgeai-example-tools.dir/WSServerSession.cpp.o -MF example/tools/CMakeFiles/edgeai-example-tools.dir/WSServerSessi
on.cpp.o.d -o example/tools/CMakeFiles/edgeai-example-tools.dir/WSServerSession.cpp.o -c TOPDIR/BUILD/work/o22-webos-linux/edgeai-vision/1.1.0-85/git/example/tools/WSServerSession.cpp
TOPDIR/BUILD/work/o22-webos-linux/edgeai-vision/1.1.0-85/git/example/tools/WSServerSession.cpp: In member function 'void aif::WSServerSession::onMessage(boost::beast::error_code, std::size_t)':
TOPDIR/BUILD/work/o22-webos-linux/edgeai-vision/1.1.0-85/git/example/tools/WSServerSession.cpp:155:26: error: 'buffer_cast' is not a member of 'boost::asio'; did you mean 'buffer_copy'?
  155 |             boost::asio::buffer_cast<char*>(m_buffer.data()), m_buffer.size());
      |                          ^~~~~~~~~~~
      |                          buffer_copy

boost::asio::buffer_cast is deprecated since boost-1.66:
https://github.com/boostorg/asio/commit/b60e92b13ef68dfbb9af180d76eae41d22e19356#diff-b393b34ba76692f3a6c00cf880fa5bda03a803f572b2575874756a8556a61dfe
and was removed in boost-1.87 with:
https://github.com/boostorg/asio/commit/7d1aac2ffd8c9e2592ac85a92be0a7528d444c8e

:Testing Performed:
Only build tested.

:QA Notes:
No change to image.

:Issues Addressed:
[WRR-17604] Fix build with boost-1.87.0
[WRR-5443] Create GPVB with Yocto 5.2 Walnascar

Change-Id: Ic2338ba3ed60bcf2dc5919dd7a14cb80ddcfc95d
---
Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/edge-ai-computer-vision/+/466657 CMakeLists.txt: allow to disable building examples to fix build with boost-1.87]

 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f094842..0ee648f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,7 +26,9 @@ ENDIF()
 add_subdirectory(core)
 add_subdirectory(extensions)
 add_subdirectory(bin)
-add_subdirectory(example)
+IF(WITH_EXAMPLES)
+    add_subdirectory(example)
+ENDIF()
 
 IF(WITH_HOST_TEST)        ### for meta-shift unit test ###
     enable_testing()
